subsys/bootloader/bl_validation/bl_validation.c

```diff
@@ -22,30 +22,52 @@ int set_monotonic_version(uint16_t version, uint16_t slot)
 	printk("Setting monotonic counter (version: %d, slot: %d)\r\n",
 		version, slot);
 
-	int err = set_monotonic_counter((version << 1) | !slot);
+	uint16_t num_cnt_slots;
+	int err;
+
+	err = num_monotonic_counter_slots(BL_MONOTONIC_COUNTERS_DESC_NSIB, &num_cnt_slots);
+	if (err != 0) {
+		printk("Failed during reading of the number of the counter slots, error: %d\r\n",
+		       err);
+		return err;
+	}
 
-	if (num_monotonic_counter_slots() == 0) {
+	if (num_cnt_slots == 0) {
 		printk("Monotonic version counter is disabled.\r\n");
-	} else if (err != 0) {
-		printk("set_monotonic_counter() error: %d\n\r", err);
+		return -EINVAL;
+	}
+
+	err = set_monotonic_counter(BL_MONOTONIC_COUNTERS_DESC_NSIB, (version << 1) | !slot);
+	if (err != 0) {
+		printk("Failed during setting the monotonic counter, error: %d\r\n", err);
 	}
 
 	return err;
 }
 
-uint16_t get_monotonic_version(uint16_t *slot_out)
+int get_monotonic_version(uint16_t *slot_out, uint16_t *version)
```

Comment:
Not a blocker: IMO this should be split into two functions now. It does two things (get the active slot and the version), and passing "NULL" as an expected argument is something one should try to avoid as the intent is unclear for a reader.

Reply:
I agree.

I suppose I figured it was out-of-scope to do a refactoring like this, but the PR is actually making things worse, so you could argue it is in-scope...

Reply:
Refactored.